libnmstate/metadata.py

```diff
@@ -40,7 +40,7 @@
 ROUTE_RULES_METADATA = "_route_rules"
 
 
-def generate_ifaces_metadata(desired_state, current_state):
+def generate_ifaces_metadata(nm_client, desired_state, current_state):
```

Comment:
In case anyone bring this up.

Yes, the metadata should not depending on NM.Client. But changing that require `libnmstate.show()` to provides  metadata on current interface holding the DNS configuration(so that we could clear it when assign DNS configure to other interface), and that change is big and could be done after this PR.

The issue is tracking at https://github.com/nmstate/nmstate/issues/972

Reply:
That is work in PR #1035 which moved the DNS stuff to plugin itself. Metadata will not depend on `NM.Client`.

I don't get the detail of `this could appear as a separate commit.`. 
When replacing global `NM.Client`, all impacted functions should using plugin level `NM.Client`.